source/blog/_posts/2017-08-24-mutation-testing.md

```diff
@@ -0,0 +1,212 @@
+---
+title: Mutation Testing
+tags:
+- phpunit
+- humbug
+- testing
+
+categories:
+- dev
+
+authors: [tn]
+github_link: blog/_posts/2017-08-24-mutation-testing.md
+
+---
+
+<div class="alert alert-info">
+Anyone who is just searching for the example source code and does not want to read the complete blog post, <a href="https://github.com/teiling88/mutation-testing">here it is.</a>
+</div>
+
+In this blog post I want to present the concept of mutation testing and a simple example with [humbug](https://github.com/humbug/humbug) for you. For a few months we have reached the 100 % code coverage goal in our actual project the [b2b-suite](https://docs.enterprise.shopware.com/b2b-suite/). But what does this number stand for? Yes it ***only*** says you created enough unit-tests to execute every single line of code in your application. Nothing more. In the following sections I will create an simple class which is completely covered with unit tests. After that we will improve the tests and show how mutation testing can support us there. 
+
+## Create an example class and unit tests
+
+Let us have a look on this simple comparison class as example:
+```php
+<?php declare(strict_types=1);
+
+class Comparison
+{
+    public function isGreaterThan(int $x, int $y): bool
+    {
+        return $x > $y;
+    }
+
+    public function isSmallerThan(int $x, int $y): bool
+    {
+        return $x < $y;
+    }
+}
+```
+
+To create an unit-test to reach 100 % coverage is very simple. See the example below: 
```

Comment:
nope, man sagt "to create **a** unit-test [...]" - unit wird praktisch 'junit' ausgesprochen, und darauf kommt's an :)